tests/integration/test_execution.py

```diff
@@ -275,6 +281,40 @@ def test_should_send_to_another_destination():
     assert message_status.details == some_body
 
 
+def test_should_use_heartbeat_and_then_lost_connection_due_message_takes_longer_than_heartbeat(caplog, settings):
+    """
+    The listener in this code has an, arguably broken, message handler (see
+    _test_callback_function_with_sleep_three_seconds function) which takes longer to process than the
+    heartbeat time of 1 second (1000); resulting in a heartbeat timeout when
+    a message is received, and a subsequent disconnect.
+    """
+    caplog.set_level(logging.DEBUG)
+    some_destination = f"some-lorem-destination-{uuid.uuid4()}"
+
+    # In order to publish sample data
+    with build_publisher().auto_open_close_connection() as publisher:
+        some_body = {"keyOne": 1, "keyTwo": 2}
+        publisher.send(some_body, some_destination, attempt=1)
+
+    settings.STOMP_OUTGOING_HEARTBIT = 1000
```

Comment:
Great test. However, due to the following STOMP protocol documentation (https://stomp.github.io/stomp-specification-1.2.html#Heart-beating):

```
because of timing inaccuracies, the receiver SHOULD be tolerant and take into account an error margin
```

Such inaccuracies can be broker-specific, such as for ActiveMQ which multiples the **heart beat time by two** as specified by theirs docs:

https://activemq.apache.org/components/artemis/documentation/2.2.0/protocols-interoperability.html

Hence, I'd just leave a side-note that this test passes for an ActiveMQ broker test.